seqr/utils/vcf_utils.py

```diff
@@ -0,0 +1,70 @@
+import re
+
+from collections import defaultdict
+
+from seqr.utils.middleware import ErrorsWarningsException
+from seqr.utils.file_utils import file_iter
+
+BLOCK_SIZE = 65536
+
+EXPECTED_META_FIELDS ={
+    'INFO': {
+        'AC': 'Integer',
+        'AN': 'Integer',
+        'AF': 'Float'
+    },
+    'FORMAT': {
+        'AD': 'Integer',
+        'DP': 'Integer',
+        'GQ': 'Integer',
+        'GT': 'String'
+    }
+}
+
+REQUIRED_HEADERS = ['#CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT']
+
+
+def _validate_vcf_header(header):
+    missing_headers = [h for h in REQUIRED_HEADERS if h not in header ]
+    if missing_headers:
+        missing_fields = ', '.join(missing_headers)
+        raise ErrorsWarningsException([f'Missing required VCF header field(s) {missing_fields}.'], [])
+
+
+def _validate_vcf_meta(meta):
+    for field, sub_field_meta in EXPECTED_META_FIELDS.items():
+        for sub_field, field_type in sub_field_meta.items():
+            if meta.get(field, {}).get(sub_field) != field_type:
+                error = f'VCF header field {field}.{sub_field} and meta information Type={field_type} is expected.'
```

Comment:
I think you want to be clear about whether the fields are missing completely or if the type is mismatched, and I think we want to make sure to send all the field errors in one validation response instead of one at at time. So we want a list of errors, one for all the missing subfields per group, and one for each mismatched type. An example of the errors we would return is this:
```
Missing required INFO field(s) AN
Missing required FORMAT field(s) AD, GT
Incorrect meta Type for INFO.AF - expected "Float", got "Integer"
Incorrect meta Type for FORMAT.GQ - expected "Integer", got "String"
Incorrect meta Type for FORMAT.DP - expected "Integer", got "Float"
```

Reply:
Okay, it is doable.